plugins/modules/net_tools/nmcli.py

```diff
@@ -1173,6 +1451,62 @@ def show_connection(self):
 
         return conn_info
 
+    def get_supported_properties(self, setting, set_property=None, set_value='FAKEVALUE'):
+        properties = []
+
+        if setting == '802-11-wireless-security':
+            set_property = set_property or 'psk'
+
+        if set_property:
+            commands = ['set %s.%s %s' % (setting, set_property, set_value)]
+        else:
+            commands = []
+
+        commands += ['print %s' % setting, 'quit', 'yes']
+        data = "\n".join(commands)
+        cmd = [self.nmcli_bin, 'con', 'edit', 'type', self.type]
+
+        (rc, out, err) = self.execute_command(cmd, data=data)
+
+        if rc != 0:
+            raise NmcliModuleError(err)
+
+        for line in out.splitlines():
+            prefix = '%s.' % setting
+            if (line.startswith(prefix)):
+                pair = line.split(':', 1)
+                property = pair[0].strip().replace(prefix, '')
+                properties.append(property)
+
+        return properties
+
+    def check_supported_properties(self, setting):
```

Comment:
Can this function be broken down further? At least the deletion of unsupported properties should be separated since that is outside the scope of "checking".

Reply:
Sure, I can mess around with that.

Reply:
Alright, I moved the `unsupported` property `deletion` into the `main` function, hopefully that at least partially satisfies your concern.